modules/data.land/R/IC_SOILGRID_Utilities.R

```diff
@@ -0,0 +1,373 @@
+        default_soilC, sum(still_missing)
+      ))
+      
+      if (verbose) {
+        for (i in which(still_missing)) {
+          PEcAn.logger::logger.debug(sprintf(
+            "Using default soil carbon value (%.2f kg/m²) for site %s", 
+            default_soilC, processed$Site_ID[i]
+          ))
+        }
+      }
+    }
+  }
+  
+  # Handle missing values in Std_soilC_0-30cm
+  na_count <- sum(is.na(processed$`Std_soilC_0-30cm`))
+  if (na_count > 0) {
+    PEcAn.logger::logger.warn(sprintf("Found %d missing values in soil carbon uncertainty", na_count))
+    
+    # Sites with missing 0-30cm but available 0-200cm uncertainty data
+    has_200cm_data <- is.na(processed$`Std_soilC_0-30cm`) & !is.na(processed$`Std_soilC_0-200cm`)
+    if (any(has_200cm_data)) {
+      processed$`Std_soilC_0-30cm`[has_200cm_data] <- processed$`Std_soilC_0-200cm`[has_200cm_data] * 0.15
+      PEcAn.logger::logger.warn(sprintf(
+        "Using scaled 0-200cm soil carbon uncertainty for %d site(s)", sum(has_200cm_data)
+      ))
+      
+      if (verbose) {
+        for (i in which(has_200cm_data)) {
+          PEcAn.logger::logger.debug(sprintf(
+            "Using scaled 0-200cm soil carbon uncertainty (%.2f) for site %s", 
+            processed$`Std_soilC_0-30cm`[i], processed$Site_ID[i]
+          ))
+        }
+      }
+    }
+    
+    # Sites still with missing uncertainty - use default percentage of mean
+    still_missing <- is.na(processed$`Std_soilC_0-30cm`)
+    if (any(still_missing)) {
+      processed$`Std_soilC_0-30cm`[still_missing] <- 
+        processed$`Total_soilC_0-30cm`[still_missing] * default_uncertainty
+      PEcAn.logger::logger.warn(sprintf(
+        "Using default uncertainty (%.1f%% of mean) for %d site(s)", 
+        default_uncertainty * 100, sum(still_missing)
+      ))
+      
+      if (verbose) {
+        for (i in which(still_missing)) {
+          PEcAn.logger::logger.debug(sprintf(
+            "Using default uncertainty (%.1f%% of mean) for site %s", 
+            default_uncertainty * 100, processed$Site_ID[i]
+          ))
+        }
+      }
+    }
+  }
+  
+  # Ensure standard deviation is non-negative 
+  neg_sd_count <- sum(processed$`Std_soilC_0-30cm` < 0, na.rm = TRUE)
+  if (neg_sd_count > 0) {
+    PEcAn.logger::logger.warn(sprintf("Found %d negative standard deviations", neg_sd_count))
+    processed$`Std_soilC_0-30cm` <- pmax(processed$`Std_soilC_0-30cm`, 0, na.rm = TRUE)
+  }
+  
+  # Ensure mean is non-negative 
+  neg_mean_count <- sum(processed$`Total_soilC_0-30cm` < 0, na.rm = TRUE)
+  if (neg_mean_count > 0) {
+    PEcAn.logger::logger.warn(sprintf("Found %d negative mean values", neg_mean_count))
+    processed$`Total_soilC_0-30cm` <- pmax(processed$`Total_soilC_0-30cm`, 0, na.rm = TRUE)
+  }
+  
+  # Add minimum standard deviation to avoid zero uncertainty 
+  min_sd <- 0.1 * processed$`Total_soilC_0-30cm` # 10% of mean as minimum SD
```

Comment:
Again, not clear that you want to make such assumptions without evidence behind it. Citation needed. Also, if there is such thing as a minimum uncertainty, it's logically inconsistent to set sites with 0% uncertainty to 10% but leave those with 1% at 1%